lib/router/src/lib.rs

```diff
@@ -102,37 +135,68 @@ impl<Req, Rec, Mk> Router<Req, Rec, Mk>
 where
     Rec: Recognize<Req>,
     Mk: Make<Rec::Target>,
-    Mk::Value: svc::Service<Req> + Clone,
+    <Mk as Make<Rec::Target>>::Value: Clone,
+    Mk::Value: svc::Service<Req>,
 {
-    pub fn new(recognize: Rec, make: Mk, capacity: usize, max_idle_age: Duration) -> Self {
-        Router {
-            inner: Arc::new(Inner {
+    pub fn new(
+        recognize: Rec,
+        make: Mk,
+        capacity: usize,
+        max_idle_age: Duration,
+    ) -> (Self, PurgeCache<Rec::Target, LoadShed<Mk::Value>>) {
+        let (cache, cache_bg) = Cache::new(capacity, max_idle_age);
+        let router = Self {
+            inner: Inner {
                 recognize,
                 make,
-                cache: Mutex::new(Cache::new(capacity, max_idle_age)),
-            }),
-        }
+                cache,
+            },
+        };
+
+        (router, cache_bg)
     }
 }
 
-impl<Req, Rec, Mk, Svc> svc::Service<Req> for Router<Req, Rec, Mk>
+impl<Req, Rec, Svc> Router<Req, Rec, FixedMake<Rec::Target, Svc>>
 where
     Rec: Recognize<Req>,
-    Mk: Make<Rec::Target, Value = Svc>,
     Svc: svc::Service<Req> + Clone,
-    Svc::Error: Into<error::Error>,
+{
+    /// A router that is created with a fixed set of known routes.
+    ///
+    /// `recognize` will only produce targets that exist in `routes`. We never
+    /// want to expire routes from the fixed set; the explicit drop of the
```

Comment:
Nit/TIOLI: I might also rephrase this to focus on the _behaviour_ of the fixed router --- something like
"Because the set of routes will not change, routes will not be expired from the router cache, and no background task is necessary to purge it." --- I think Oliver's suggestion that we comment about the explicit drop of the purge task is more of an implementation detail that I'd put in the function body.

This is a nit & non-blocking, I think the comment is okay as-is.